infra/build/functions/ci.py

```diff
@@ -0,0 +1,260 @@
+                      help='Sanitizers.')
+  parser.add_argument('--fuzzing-engines',
+                      required=True,
+                      nargs='+',
+                      help='Fuzzing engines.')
+  parser.add_argument('--branch',
+                      required=False,
+                      default=None,
+                      help='Use specified OSS-Fuzz branch.')
+  return parser.parse_args()
+
+
+def get_production_build_statuses():
+  """Returns the status of the last build done in production for each
+  project."""
+  statuses = _get_production_build_statuses('fuzzing')
+  statuses = _get_production_build_statuses('coverage', statuses)
+  return statuses
+
+
+def get_projects(specified_projects, project_statuses, build_type):
+  """Returns the list of projects that should be built based on the projects
+  specified by the user (|specified_projects|) the |project_statuses| of the
+  last builds and the |build_type|."""
+  buildable_projects = []
+
+  for project in specified_projects:
+    if project not in project_statuses:
+      buildable_projects.append(project)
+      continue
+    if project_statuses[project].production_build[build_type]:
+      buildable_projects.append(project)
+      continue
+
+    logging.info('Skipping %s, last build failed.', project)
+
+  return buildable_projects
+
+
+def _do_build(args, config, credentials, build_type, project_statuses):
+  """Does test builds of the type specified by |build_type|."""
+  if build_type == 'fuzzing':
+    get_build_steps_func = build_project.get_build_steps
+    sanitizers = args.sanitizers
+  else:
+    get_build_steps_func = build_and_run_coverage.get_build_steps
+    sanitizers = ['coverage']
+
+  projects = get_projects(args.projects, project_statuses, build_type)
+  for project_name in projects:
+    logging.info('Getting steps for: "%s".', project_name)
+    try:
+      project_yaml, dockerfile_contents = (
+          build_project.get_project_data(project_name))
+    except FileNotFoundError:
+      logging.error('Couldn\'t get project data. Skipping %s.', project_name)
+      continue
+
+    project_yaml['sanitizers'] = list(
+        set(project_yaml['sanitizers']).intersection(set(sanitizers)))
+
+    project_yaml['fuzzing_engines'] = list(
+        set(project_yaml['fuzzing_engines']).intersection(
+            set(args.fuzzing_engines)))
+
+    if not project_yaml['sanitizers'] or not project_yaml['fuzzing_engines']:
+      logging.info('Nothing to build for this project: %s.', project_name)
+      continue
+
+    steps = get_build_steps_func(project_name, project_yaml,
+                                 dockerfile_contents, IMAGE_PROJECT,
+                                 BASE_IMAGES_PROJECT, config)
+    if not steps:
+      logging.error('No steps. Skipping %s.', project_name)
+      continue
+
+    project_statuses[project_name].set_build_id(
+        build_project.run_build(project_name, steps, credentials, build_type),
```

Comment:
One more thing: Can we tag these builds differently on cloudbuild, so we can easily filter them in the dashboard if we need to?